includes/handle.inc

```diff
@@ -154,17 +154,20 @@ function islandora_handle_append_handle(AbstractObject $object, $datastream, $xs
   if ($did_load_ok) {
     $xslt->importStylesheet($xsl_transform);
     $xslt->setParameter('', 'handle_value', $handle_url);
-    $ds_output = $xslt->transformToXml($input);
+    $ds_output = $xslt->transformToDoc($input);
     if ($ds_output) {
-      $object[$datastream]->content = $ds_output;
-      return array(
-        'success' => TRUE,
-        'message' => array(
-          'message' => t('Appended Handle to @ds datastream for @pid!'),
-          'message_sub' => array('@pid' => $object->id, '@ds' => $datastream),
-          'type' => 'dsm',
-        ),
-      );
+      // If there is nothing to change don't make a new version.
+      if (!islandora_handle_datastream_content_equal($input, $ds_output)) {
```

Comment:
Had a quick chat w/ @jordandukart: We wouldn't report anything (success, failure or otherwise) in the event we do not set the content... We're okay with this.